### ISSUE TITLE
`joi` should be declared as a dependency?

### ISSUE THREAD

**Issue by dominykas** [hapijs/hapi#4536]
### Runtime

node.js

### Runtime version

v22

### Module version

latest (23.x)

### What are you trying to achieve or the steps to reproduce?

Use `@hapi/hapi` in a Typescript project without having `joi` installed.

### What was the result you got?

```
node_modules/@hapi/hapi/lib/types/route.d.ts:2:68 - error TS2307: Cannot find module 'joi' or its corresponding type declarations.

2 import { ObjectSchema, ValidationOptions, SchemaMap, Schema } from 'joi';
                                                                     ~~~~~

node_modules/@hapi/hapi/lib/types/server/server.d.ts:4:22 - error TS2307: Cannot find module 'joi' or its corresponding type declarations.

4 import { Root } from 'joi';
```

### What result did you expect?

No compilation errors.

**Comment by kanongil** [comment]
This is a duplicate of #4491. There is already a pending PR to fix this in #4497.

Note that you can fix this locally by installing the `joi` dependency yourself, or adding `skipLibCheck: false` to your tsconfig (with other side effects).

**Comment by dominykas** [comment]
Sorry, missed those open issues 🤦
